pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -131,11 +140,16 @@ func (a *index) constructServiceEntries(svc *networkingclient.ServiceEntry, w *W
 	// TODO this is only checking one controller - we may be missing service vips for instances in another cluster
 	res := make([]*workloadapi.Service, 0, len(svc.Spec.Hosts))
 	for _, h := range svc.Spec.Hosts {
+		// if we have no user-provided addresses and h is not wildcarded we can try to use autoassigned addresses
+		a := addresses
+		if len(a) == 0 && !host.Name(h).IsWildCarded() {
```

Comment:
Is this a desired behavior we can fence with a unit test?

Reply:
added testing for multiple different opt-out scenarios and an opt-in